### PR TITLE
[discover] Clean up enhanced search API

### PR DESCRIPTION
**Pull request by kavilla**
### Description

* Deletes the connections service.
* Removes dataframe object being passed into request
* Deletes unused APIs
* Error handling

### Issues Resolved

n/a

## Screenshot

Screenshot 2024-09-18 at 2 17 17 PM

## Testing the changes

locally

## Changelog

- fix: Remove unused enhancements APIs, clean up, and error handling

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by abbyhu2000 — APPROVED** [review]

**Review by sejli — APPROVED** [review]

File: src/plugins/query_enhancements/public/datasets/s3_type.ts [hunk 1]
```diff
@@ -171,9 +174,13 @@ const fetchConnections = async (
   http: HttpSetup,
   dataSource: DataStructure
 ): Promise<DataStructure[]> => {
-  const query = (dataSource.meta as DataStructureCustomMeta).query;
-  const response = await http.fetch(`../../api/enhancements/datasource/external`, {
-    query,
+  const abortController = new AbortController();
```
Comment (ashwin-pc) [inline src/plugins/query_enhancements/public/datasets/s3_type.ts:177]: What do we do with this abort controller?

File: src/plugins/data/common/constants.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const DEFAULT_DATA = {
     } as DataStructure,
     LOCAL_DATASOURCE: {
       id: '',
-      title: 'Local Cluster',
+      title: 'Default Cluster',
```
[the remark this answers is not in the document]
Comment (kavilla) [reply src/plugins/data/common/constants.ts:47]: yeah that makes sense i was worried making too big of a change in the case of no local cluster.

should we create an issue for this?

File: src/plugins/query_enhancements/public/datasets/s3_type.ts (hunk 1 above)
Comment (kavilla) [reply src/plugins/query_enhancements/public/datasets/s3_type.ts:177]: need to verify if using it correctly yet. but in the case of the user selecting a connection accidentally and then selecting another connection immediately aborting the previous call

File: src/plugins/data/common/constants.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply src/plugins/data/common/constants.ts:47]: Yes, the expected behaviour is that this option only shows up when you have a local cluster. If MDS is off the index list should be shown directly.

**Pull request merged**